web/app/scripts/src/directives/proportion-field.es.js

```diff
@@ -0,0 +1,55 @@
+import { angular } from '../libraries';
+
+import ProportionField from '../services/proportion-field';
+
+const moduleProperties = {
+    moduleName: 'incuna-surveys.proportion-field-directive'
+};
+
+const module = angular.module(moduleProperties.moduleName, [
+    ProportionField.moduleName
+]);
+
+module.directive('proportionField', [
+    ProportionField.componentName,
+    function (ProportionField) {
+        return {
+            restrict: 'A',
+            scope: {
+                model: '=',
+                options: '=proportionField'
+            },
+            templateUrl: 'templates/incuna-surveys/form/proportion-field.html',
+            link: function (scope) {
+                scope.fields = [];
+                scope.$watch('options', (options) => {
```

Comment:
That's a good point, but perhaps you would want the fields to rebuild after an options change. However, this will only occur if the options thing was replaced, not if the object was altered, so practically this should only run once when `scope.options === undefined` becomes `scope.options === boundObject`